mirgecom/boundary.py

```diff
@@ -138,4 +166,39 @@ def boundary_pair(self, discr, q, btag, **kwargs):
                                     momentum=ext_mom,
                                     species_mass=int_cv.species_mass)
 
-        return TracePair(btag, interior=int_soln, exterior=bndry_soln)
+        return bndry_soln
+
+    def exterior_grad_q(self, discr, grad_q, btag, **kwargs):
+        """Get the exterior grad(Q) on the boundary."""
+        # Grab some boundary-relevant data
+        num_equations, dim = grad_q.shape
+        num_species = num_equations - dim - 2
+        cv = split_conserved(dim, grad_q)
+        actx = cv.mass[0].array_context
+        # Grab a unit normal to the boundary
+        normal = thaw(actx, discr.normal(btag))
+
+        # Get the interior soln
+        int_soln = discr.project("vol", btag, grad_q)
+        int_cv = split_conserved(dim, int_soln)
```

Comment:
I noticed this on the other PR, but I wasn't sure whether to comment there or here. The naming of `int_soln` and `int_cv` here is confusing IMO.

Reply:
Heh, yeah.  What about this:
```python
facial_gradq = discr.project("vol", btag, grad_q)
gradq_cv = split_conserved(dim, facial_gradq)
```
The only reason I didn't go with something like `gradq_components` is due to the unwieldy length of it.

Reply:
Either sounds good. I used `*_int` in the diffusion BCs: https://github.com/illinois-ceesd/mirgecom/blob/main/mirgecom/diffusion.py#L145

Reply:
w-hagen/mirgecom#17